src/index.js

```diff
@@ -193,6 +210,54 @@ class DropdownTreeSelect extends Component {
     this.keepDropdownActive = false
   }
 
+  onKeyboardKeyDown = e => {
+    const { readOnly, simpleSelect } = this.props
+    const { showDropdown, tags, searchModeOn, currentFocus } = this.state
+    const tm = this.treeManager
+    const tree = searchModeOn ? tm.matchTree : tm.tree
+
+    if (!showDropdown && (keyboardNavigation.isValidKey(e.key, false) || /^\w$/i.test(e.key))) {
+      // Triggers open of dropdown and retriggers event
+      e.persist()
+      this.handleClick(null, () => this.onKeyboardKeyDown(e))
```

Comment:
Help me understand why we're doing it this way - especially why we're calling `onKeyboardKeyDown` again within the `onKeyboardKeyDown` handler.

I believe you're trying to reuse the state setting logic and then react after that logic does its thing?  If so, I think it's time to refactor `handleClick`.

Reply:
I wait for the state to update and retrigger the event to select the first node in the now visible dropdown.